libraries/Wire/src/utility/twi.c

```diff
@@ -396,6 +424,40 @@ void twi_releaseBus(void)
   twi_state = TWI_READY;
 }
 
+/* 
+ * Function twi_setTimeoutInMicros
+ * Desc     set a global timeout for while loops that we might get stuck in
+ * Input    timeout value in microseconds
+ * Output   none
+ */
+void twi_setTimeoutInMicros(uint32_t timeout)
+{
+  twi_timeout_us = timeout;
+}
+
+/* 
+ * Function twi_handleTimeout
+ * Desc     do this stuff when a while loop here has run for longer than twi_timeout_us microseconds
+ * Input    none
+ * Output   none
+ */
+void twi_handleTimeout(void)
+{
+  // remember the bitrate register value
+  uint8_t previous_TWBR = TWBR;
+
+  // remember the address register value
+  uint8_t previous_TWAR = TWBR;
```

Comment:
```suggestion
  // remember bitrate and address settings
  uint8_t previous_TWBR = TWBR;
  uint8_t previous_TWAR = TWAR;
```

Reply:
(note that I changed the comment and layout, but there was also a TWAR/TWBR mixup in there)

Reply:
Oh! Nice catch!